examples/opup/contract.py

```diff
@@ -0,0 +1,35 @@
+from pyteal import *
+from typing import Literal
+from beaker.contracts import OpUp
+from beaker.decorators import bare_handler, handler
+
+
+class ExpensiveApp(OpUp):
+    @handler
+    def bootstrap(self, ptxn: abi.PaymentTransaction, *, output: abi.Uint64):
+        return Seq(
+            Assert(ptxn.get().amount() >= OpUp.min_balance),
+            ExpensiveApp.create_opup(),
+            output.set(ExpensiveApp.opup_app_id),
+        )
+
+    @handler
+    def hash_it(input: abi.String, iters: abi.Uint64, opup_app: abi.Application, *, output: abi.String):
```

Comment:
how do we convey that the caller must pass the opup app id?

Reply:
some argument to the handler to specify the requirement? `@handler(require_arg=RequiredArg(opup_app=OpUp.opup_app_id)))` I dont like this but it would provide us the chance to give some more context to the caller